mitsurunner.yaml

```diff
@@ -81,96 +90,172 @@ sensor:
         float temperature_delta = (id(outdoor_temp).state - id(heat_exchanger_temp).state);
         bool reset_min_time_of_temperature_delta = false;
 
-        // if temperature delta is over treshold and minimum heating time has passed
-        if ( temperature_delta >= TEMPERATURE_DELTA_TO_DEFROST && id(G_min_heating_time_passed) == true) {
-            // if temperature has been long enough over treshold or force defrost temperature is exceeded
-            if ( id(G_excess_min_time_passed) == true || temperature_delta >= TEMPERATURE_DELTA_TO_FORCE_DEFROST) {
-                // activate defrosting
-                id(trigger_defrosting).execute();
-                reset_min_time_of_temperature_delta = true;
-                }
-            // if excess timer were not yet passed, do nothing
-            else if ( id(start_excess_min_time_timer).is_running() ) {
-                }
-            // excess min time were not on, but other defrosting requirements were fullfilled to start it
-            else  
-                { 
-                id(start_excess_min_time_timer).execute();
-                }
-            }            
-        else {        
-            // if during the excess timer, the temperature returns back below treshold, then cancel the timer
-            reset_min_time_of_temperature_delta = true;       
-            }
-        
-        if ( reset_min_time_of_temperature_delta ) {
-            id(G_excess_min_time_passed) = false;
-            
-            // timer is needed to be stopped only if it is running
-            if ( id(start_excess_min_time_timer).is_running() ) {
-                id(start_excess_min_time_timer).stop();  
-                }
-            id(mqtt_client).publish(TOPIC_IS_EXCESS_MIN_TIME_PASSED, to_string(0.0));
-            }
 
         // if you want to get MQTT logs from temperature delta all the time, define TOPIC_TEMP_DELTA in constants.h file
         #ifdef TOPIC_TEMP_DELTA_MESSAGES
         id(mqtt_client).publish(TOPIC_TEMP_DELTA, to_string(temperature_delta));
         #endif
 
+        id(mqtt_client).publish(TOPIC_STATE, to_string(id(G_state)));
+
+        switch(id(G_state)) {
+
+            // This state is executed only in reset
+            case ST_RESET:
+                // If defrosting is going on
+                if (temperature_delta < 0.0) {
```

Comment:
This should be changed to temperature_delta be <= 0.0
Due to, if difference is zero, basically it means even two things: 
The pump is not on --> on harm
or then the defrozing is even starting or stopping. As it does not become clear without re-measuring, I would propose to even re-measure (minor priority,  as so small opportunity for this case) or then change as proposed as it is not so serious to give pump work as is when defrozing is starting than switching resistor to provide wrong temperature when defozing is starting. If defrozing is ending, there is no big harm from either state of the realay.

Reply:
This needs anyway more research. I've created an issue about this. I have tested that a couple of times by rebooting during defrosting, but it didn't work. But it did work if I put negative delta in the code. Maybe it didn't get proper sensor data.